### PR TITLE
Update documentation on how to initialise PySpark using hooks

### PR DESCRIPTION
**Pull request by SajidAlamQB**
Signed-off-by: SajidAlamQB <90610031+SajidAlamQB@users.noreply.github.com>

## Description

Follows on from: https://github.com/kedro-org/kedro-starters/pull/102
Related Issue: related Issue: https://github.com/kedro-org/kedro/issues/1563

Updating documentation on how PySpark is now initialised via hooks in the starters.

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro/pull/1825"></a>

### CODE REVIEW

**Comment by SajidAlamQB** [comment]
I'm wondering if it would be better to have both the context class and hooks example in the documentation instead of just replacing it with the hooks example.

File: docs/source/tools_integration/pyspark.md [hunk 1]
```diff
@@ -15,49 +15,32 @@ spark.scheduler.mode: FAIR
 Optimal configuration for Spark depends on the setup of your Spark cluster.
 ```
 
-## Initialise a `SparkSession` in custom project context class
+## Initialise a `SparkSession` using hooks
```
Comment (AntonyMilneQB) [inline docs/source/tools_integration/pyspark.md:18]:
```suggestion
## Initialise a `SparkSession` using a hook
```

File: docs/source/tools_integration/pyspark.md [hunk 2]
```diff
@@ -15,49 +15,32 @@ spark.scheduler.mode: FAIR
...
 
-Before any `PySpark` operations are performed, you should initialise your [`SparkSession`](https://spark.apache.org/docs/latest/sql-getting-started.html#starting-point-sparksession) in your custom project context class, which is the entrypoint for your Kedro project. This ensures that a `SparkSession` has been initialised before the Kedro pipeline is run.
+Before any `PySpark` operations are performed, you should initialise your [`SparkSession`](https://spark.apache.org/docs/latest/sql-getting-started.html#starting-point-sparksession) using [hooks](../hooks/introduction) in your custom project with the `after_context_created` hook specification. This ensures that a `SparkSession` has been initialised before the Kedro pipeline is run.
```
Comment (AntonyMilneQB) [inline docs/source/tools_integration/pyspark.md:20]:
```suggestion
Before any `PySpark` operations are performed, you should initialise your [`SparkSession`](https://spark.apache.org/docs/latest/sql-getting-started.html#starting-point-sparksession) using an `after_context_created` [hook](../hooks/introduction). This ensures that a `SparkSession` has been initialised before the Kedro pipeline is run.
```

File: docs/source/tools_integration/pyspark.md [hunk 3]
```diff
@@ -71,12 +54,12 @@ Call `SparkSession.builder.getOrCreate()` to obtain the `SparkSession` anywhere
 
 We don't recommend storing Spark session on the context object, as it cannot be serialised and therefore prevents the context from being initialised for some plugins.
 
-Now, you need to configure Kedro to use `CustomContext`. Set `CONTEXT_CLASS` in `src/<package_name>/settings.py` as follows:
+You will also need to register `Hooks` implementation by updating the HOOKS variable in `src/<package_name>/settings.py` as follows:
```
Comment (AntonyMilneQB) [inline docs/source/tools_integration/pyspark.md:57]:
```suggestion
You will also need to register `SparkHooks` by updating the `HOOKS` variable in `src/<package_name>/settings.py` as follows:
```

**Review by AntonyMilneQB — APPROVED** [review]
Just a few small suggestions but generally looks great, thanks! I think we should just use the hooks approach as you have it now, and not mention the custom context approach at all. In general I think we should try to move away from injecting custom behaviour using `KedroContext` in favour of hooks.

**Review by noklam — APPROVED** [review]
LGTM

**Pull request merged**